### DISCUSSION TITLE
Typing - Type Hints with MyPy

### DISCUSSION THREAD

**Discussion by pedroimpulcetto** [asacristani/fastapi-rocket-boilerplate#43] [Ideas]
Hey all, I would love to discuss if we can implement Type Hints properly.

#### Using it the right way we can have:
- **Early Bug Detection**: By running MyPy, you can catch compatibility and type mismatch errors before they cause problems at runtime, potentially saving a lot of debugging time.
- **Encourages Better Practices**: Using MyPy encourages developers to think more about the types of their variables and function return values, which can lead to better, more maintainable code.
- **Integration with Development Processes**: MyPy can be integrated into the development workflow, including continuous integration (CI) systems. This ensures that type checks are performed automatically, for instance, before code is merged into the main branch of your project.

[Typing Package](https://docs.python.org/3/library/typing.html)
[MyPy](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

**Comment by pedroimpulcetto** [comment]
What do you think @asacristani?

**Reply by asacristani** [reply]
Mypy is in the list of linters of the pre commit and CI using Trunk check.
What you have checked is a lot of warnings before adding Trunk check.
You can run `trunk check -a` and see everything we should fix and correct. If you want to collaborate here, create a new PR and fix some (of all) warnings.
Thanks @pedroimpulcetto

**Reply by asacristani** [reply]
Check this issue: https://github.com/asacristani/fastapi-rocket-boilerplate/issues/34

**Reply by pedroimpulcetto** [reply]
hum, got it! I didn't see that issue, sorry!
I definitely will work on that

btw, I have this [PR](https://github.com/asacristani/fastapi-rocket-boilerplate/pull/42) #42 fixing some types, can you check, please @asacristani ?

Thank you!
